### ISSUE TITLE
Feature request: implement UI3 front end (opensource blue iris front end)

### ISSUE THREAD

**Issue by R5fan** [blakeblackshear/frigate#1489]
I still use both BI and frigate, and I notice many people do. Blue iris for the 24/7 recording and particularly the web interface, frigate for object detection and better integration with HA. The web front end of Blue Iris turns out to be open source:

https://github.com/bp2008/ui3

I have no idea how feasible it is to port that to frigate, but that web interface is absolutely outstanding, extremely flexible, buttersmooth on mobile or desktop, very responsive, and its one of the only reasons Im still using BI. Porting it to frigate and HA could also help solve a long standing problem with Home Assistant front end being terrible at playing h264 streams (long initial delay, that creeps up over time). In fact I use BI to provide an mjpeg stream for my HA frontend, which is kinda insane.

**Comment by blakeblackshear** [comment]
I don't think porting this frontend over is something I am interested in. I would be very appreciative if you could describe specific use cases that you have for the frontend where you find UI3 meets your needs well. Screenshots to show what you like and why you like it would be great.

**Comment by scho0ck** [comment]
I'm currently in search for a proper web frontend as well the reason being that the HA camera entity is very slow and the frigate web itself not very mobile friendly also it only shows the 5fps detect stream.

At the moment I'm having blueiris and UI3 running mainly for live viewing cameras it would be great if we could improve on that directly in frigate.

**Comment by scooper1** [comment]
Do you know that you can define a ha camera entry using the frigate Web API.   I find this good for real time viewing even on mobile  even works nicely with current Google hub software...

**Comment by scho0ck** [comment]
> Do you know that you can define a ha camera entry using the frigate Web API. I find this good for real time viewing even on mobile even works nicely with current Google hub software...

Would you mind sharing how you did that ?

**Comment by scooper1** [comment]
```
ffmpeg:
stream:
camera:
  - platform: mjpeg
    mjpeg_url: http://172.16.10.98:5000/api/f_back?h=720&motion=0&bbox=1
    name: back
    authentication: basic
    verify_ssl: false

```

**Comment by R5fan** [comment]
> in. I would be very appreciative if you could describe specific use cases that you have for the frontend where you find UI3 meets your needs well. Screenshots to show what you like and why you like it would be great.

Honestly the list is endless and will be different for everyone. BI has evolved over decades (I presume) to cater for just about any use case. But Ill name some that apply to me:

- a compact event list (that can be filtered in a million ways) in the side bar that you can scroll through with icons showing what triggered the event and when. Hovering over the event gives a popup preview of the clip, clicking on an event shows the event on the rest of the screen without closing the event list (this is rather annoying in frigate currently, you watch an event, and you need to "find" it again in the list if you want to check the next or something else)

- time line that can be scrubbed and played back at 1x, 2x to 256x speed. Clicking an event in the event list jumps to that point on the time line.

- zooming in by using the scroll wheel or pinch to zoom. 

- Grouping of camera's (eg front, back, inside) and showing a mostly (or entirely) full screen grid with only camera's of that group. Clicking a camera makes that camera full screen and it switches from sub to main stream. Seamlessly (ie, it play the sub stream full screen until the main stream is ready) 
 
- configurable stream profiles (resolution, bitrate, ..)

- Exporting of arbitrary segments

The biggest thing however, its incredibly smooth and responsive. I made a video to give you an idea, and that also shows most of the above functions (but only fraction of what it can do, others will use other functions for sure) . If you havent used BI (which I am assuming), then be assured this is indeed the webinterface, not the windows gui, and it works just as well on my phone:

https://youtu.be/jBOPepQpRwU

The backend isnt particularly powerful, BI runs on a 2 GHz core i3 under proxmox and its given 1 core and 4GB ram.

**Comment by R5fan** [comment]
> ```
> ffmpeg:
> stream:
> camera:
>   - platform: mjpeg
>     mjpeg_url: http://172.16.10.98:5000/api/f_back?h=720&motion=0&bbox=1
>     name: back
>     authentication: basic
>     verify_ssl: false
> ```

That works well because its mjpeg.  Mjpeg streams play fine on HA front end, but h264/5 do not. Mjpeg is incredibly bandwidth hungry. May not be a problem on your local lan or wifi, but I maxed out my monthly mobile data plan in 20 minutes before I limited resolution to VGA and frame rate to 1 FPS, and before HA companion properly closed the connections in the background.
